gate-proxy/src/main/kotlin/com/netflix/spinnaker/gate/controllers/ProxyController.kt

```diff
@@ -40,40 +40,70 @@ import org.springframework.web.bind.annotation.RequestMethod.PUT
 import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
 import org.springframework.web.servlet.HandlerMapping
+import javax.annotation.PostConstruct
 
 @RestController
 @RequestMapping(value = ["/proxies"])
-class ProxyController(
+public class ProxyController(
   val objectMapper: ObjectMapper,
   val registry: Registry,
-  val proxyConfigurationProperties: ProxyConfigurationProperties
+  val proxyConfigProviders: List<ProxyConfigProvider>
```

Comment:
Ah - we will need to wrap this in `ObjectProvider` and `Provider`.  This is because the v2 plugin framework does not make guarantees about bean ordering.

Reply:
Makes sense, I'll follow-up on this once I add an actual plugin implementing this extension point.